src/storage/exec/GetNeighborsNode.h

```diff
@@ -82,14 +82,24 @@ class GetNeighborsNode : public QueryNode<VertexID> {
       row[1].setList(agg->mutableResult().moveList());
     }
 
-    resultDataSet_->rows.emplace_back(std::move(row));
+    if (!context_->filterInvalidResultOut || context_->resultStat_ == ResultStatus::NORMAL) {
```

Comment:
I am not pretty sure about whether we need to check `resultStat_` here. IIRC, we should not check it. You could check it with two edges, one can pass, another can't.

Reply:
talked offline.